doc/installing.rst

```diff
@@ -4,182 +4,181 @@ Installation
-        cd PypeIt
+        conda env create -f environment.yml
 
- #. Install dependencies
+ #. Activate it::
 
-  - into your current environment with ONE of the following commands (you may need to also install
-    PySide2/PyQt5 if they are not already installed as per note above)::
+        conda activate pypeit
 
-        pip install -r pypeit/requirements.txt
+ #. Verify that the new environment was installed correctly::
 
-        conda install -y -c conda-forge --file requirements.txt
+        conda env list
 
-  - or into a new environment::
+ #. Install latest ``pypeit`` via ``pip`` as above or perform a developer install as below or install the latest
+ release from ``conda-forge``::
+
+        conda install -c conda-forge pypeit
+
+#. Install the preferred QT binding either via ``pip`` as above or via conda::
+
+        conda install -c conda-forge pyqt
+
+or
 
-        conda env create -f environment.yml
-        conda activate pypeit
-        # install PySide2/PyQt5 from conda/pip using using ONE of the following lines
         conda install -c conda-forge pyside2
-        conda install pyqt
-        pip install pyside2
-        pip install pyqt5
 
- #. Install PypeIt::
+Developer Install via ``pip``
+-----------------------------
+
+It is also possible to install pre-release or development versions of ``PypeIt`` directly from `GitHub <https://github.com/pypeit/PypeIt>`_
+using ``pip``. If you already have a ``pypeit`` environment setup, do:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt#egg=pypeit
+
+If you're installing in a clean environment, be sure to include the optional dependencies as well:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt#egg=pypeit[pyqt5,pyside2,shapely]
+
+Those commands will install the default branch, ``master``. You can also specify the branch you wish to use:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt.git@develop#egg=pypeit[pyqt5,pyside2,shapely]
+
+Commit hashes, tag names, or git refs can also be specified. See the `VCS Support documentation
+<https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support>`_ for details and examples.
+
+Developers doing code development will likely want to set up an "editable" install that points to a locally checked out
+copy of the GitHub repository. It is highly recommened to use ``pip`` for this as well so that the dependencies can be managed.
+It is also recommended to install all optional dependencies within the environment used for ``PypeIt`` development. First,
+clone the repository:
+
+        git clone https://github.com/pypeit/PypeIt.git
 
-        pip install -e .
+Then perform the install, preferably in a clean environment:
 
-Installing the code this way ensures that virtually all changes to files in
-the ``PypeIt`` directory take immediate effect the next time you
-import the code.
+        cd PypeIt
+        pip install -e .[pyqt5,pyside2,shapely,test,docs]
```

Comment:
Same questions as in the development doc: Do we need both `pyqt5` and `pyside2` in this call, and can we consolidate all of these (as well as keeping them separate) into a `dev` option?

Reply:
now that i have `dev` set up, i'll update this to use that.